src/RelayServer.ts

```diff
@@ -527,35 +492,43 @@ export class RelayServer extends EventEmitter {
 
     await validateIfGasAmountIsAcceptable(envelopingTransaction);
 
-    const maxPossibleGas = await this.getMaxPossibleGas(envelopingTransaction);
+    const { maxPossibleGas, maxPossibleGasWithFee } =
+      await this.getMaxPossibleGas(envelopingTransaction);
 
     // Send relayed transaction
     log.debug('maxPossibleGas is', maxPossibleGas.toString());
 
     await validateIfTokenAmountIsAcceptable(
-      maxPossibleGas,
+      maxPossibleGasWithFee,
       envelopingTransaction,
       this.config.app
     );
 
     const relayHub = getRelayHub();
 
-    const method = isDeployTransaction(envelopingTransaction)
-      ? await relayHub.populateTransaction.deployCall(
-          envelopingTransaction.relayRequest,
-          envelopingTransaction.metadata.signature
-        )
+    const {
+      relayRequest,
+      metadata: { signature },
+    } = envelopingTransaction;
+    const {
+      relayData: { gasPrice },
+    } = relayRequest;
+
+    const method = isDeployRequest(relayRequest)
+      ? await relayHub.populateTransaction.deployCall(relayRequest, signature)
       : await relayHub.populateTransaction.relayCall(
-          envelopingTransaction.relayRequest as RelayRequest,
-          envelopingTransaction.metadata.signature
+          relayRequest as RelayRequest,
+          signature
         );
 
     // Call relayCall as a view function to see if we'll get paid for relaying this tx
-    const maxPossibleGasWithViewCall = await this.maxPossibleGasWithViewCall(
-      method,
-      envelopingTransaction,
-      maxPossibleGas
```

Comment:
@franciscotobar It looks like we were wrongly using `maxPossibleGasWithFee`.

Reply:
I think so, Im not pretty sure about the names that I assigned, but to keep the same behavior as the client I return both values.